### ISSUE TITLE
false positive messages on legal errors from the same file

### ISSUE THREAD

**Issue by ahlininv** [tomarrell/wrapcheck#6]
file **internal/modifier/modifiers_aggregate.go** contains code:

```go
var (
    ErrPlaylistsNotTwo = errors.New("playlists not two")
)

if len(playlists) != 2 {
    return nil, ErrPlaylistsNotTwo
}
```

**causes the following:**
internal/modifier/modifiers_aggregate.go:197:15: error returned from external package is unwrapped (wrapcheck)
                return nil, ErrPlaylistsNotTwo

**Comment by tomarrell** [comment]
Thanks for the report.

Hmm I can see why that might be an issue. If the same error is returned in multiple places however, then it becomes a little tricky to hunt down. Obviously that would be the fault of the developer though.

Maybe something like this would be good to put behind a flag, as I think your use-case is valid. I'll have a think about it and come up with something in an upcoming release for this.

**Comment by tomarrell** [comment]
Hey @ahlininv ,

I've been taking a look into this to reproduce it, however I'm not able to with the following test case:

```go
package main

import "errors"

func main() {
	do()
}

func do() (*string, error) {
	var (
		ErrPlaylistsNotTwo = errors.New("playlists not two")
	)

	if true {
		return nil, ErrPlaylistsNotTwo
	}

	return nil, nil
}
```

Would you be able to provide any further reproduction steps?

**Comment by jetexe** [comment]
> Would you be able to provide any further reproduction steps?

Just change `import "errors"` to `import "github.com/pkg/errors"`

```go
package main

import "github.com/pkg/errors"

func main() {
	_ = do()
}

func do() error {

	return errors.New("some error")
}

```

**Comment by tomarrell** [comment]
This has been resolved as part of `v1.1.0`. It has already been merged to master of golangci-lint as well and should be available in the `1.40.0` release.

Cheers!
